lib/spack/docs/build_systems/intelpackage.rst

```diff
@@ -0,0 +1,607 @@
+`Build customization <http://spack.readthedocs.io/en/latest/build_settings.html>`_.
+
+Similar to ``compilers.yaml``, the ``packages.yaml`` files define a package
+external to Spack in terms of a Spack spec and resolve each such spec via
+either the ``paths`` or ``modules`` tokens to a specific pre-installed package
+version on the system.  Since Intel tools generally need environment variables
+to interoperate, which cannot be conveyed in a mere ``paths`` specification,
+the ``modules`` token will be more sensible to use. It resolves the Spack-side
+spec to a modulefile generated and managed outside of Spack's purview,
+which Spack will load internally and transiently when the corresponding spec is
+called upon to compile client packages.
+
+If your system administrator did not provide modules for pre-installed Intel
+tools, you could do well to ask for them, because installing multiple copies
+of the Intel tools, as is wont to happen once Spack is in the picture, is
+bound to stretch disk space and patience thin. If you *are* the system
+administrator and are still new to modules, then perhaps it's best to follow
+the `next section <Installing Intel tools within Spack_>`_ and install the tools
+solely within Spack.
+
+The following example integrates two packages embodied by hypothetical
+external modulefiles ``intel-mkl/18/18.0.1`` and ``intel-mkl/18/18.0.2``, as
+Spack packages ``intel-mkl@2018.1.163`` and ``intel-mkl@2018.2.199``,
+respectively.
+
+.. code-block:: yaml
+
+   packages:
+     intel-mkl:
+       modules:
+         intel-mkl@2018.1.163  arch=linux-centos6-x86_64:  intel-mkl/18/18.0.1
+         intel-mkl@2018.2.199  arch=linux-centos6-x86_64:  intel-mkl/18/18.0.2
+
+Note that the version numbers in the ``intel-mkl`` spec correspond to the ones
+used for the Intel products and adopted within Spack. You can inspect them by:
+
+.. code-block:: sh
+
+  spack info intel-mkl
+
+Using the same version numbers is useful for clarity, but not strictly necessary.
+
+Note that the Spack spec in the example does not contain a compiler
+specification. This is intentional, as the Intel library packages can be used
+unmodified with different compilers.
+
+**TODO:** Confirm how the compiler-less spec is handled.
+
+A slightly more advanced example follows, illustrating how to provide
+`variants <http://spack.readthedocs.io/en/latest/basic_usage.html#variants>`_
+and using the ``buildable: False`` directive to prevent Spack from installing
+other versions or variants of the named package through its normal internal
+mechanism.
+
+.. code-block:: yaml
+
+   packages:
+     intel-parallel-studio:
+       modules:
+         intel-parallel-studio@cluster.2018.1.163 +mkl+mpi+ipp+tbb+daal  arch=linux-centos6-x86_64:  intel/18/18.0.1
+         intel-parallel-studio@cluster.2018.2.199 +mkl+mpi+ipp+tbb+daal  arch=linux-centos6-x86_64:  intel/18/18.0.2
+       buildable: False
+
+**TODO:** Confirm variant handling.
+
```

Comment:
maybe add here no-module config as an example? That's how I configure:
```
intel-mpi:
    version: [2018.2.199]
    paths:
      intel-mpi@2018.2.199%intel@18.0.2: /apps/intel/ComposerXE2018/
    buildable: False
  intel-mkl:
    version: [2018.2.199]
    paths:
      intel-mkl@2018.2.199%intel@18.0.2: /apps/intel/ComposerXE2018/
    buildable: False
```
Probably one could drop `%intel@18.0.2`

Reply:
Good idea as well, for folks _sans_ modules, be it intentionally or unfortunately.

Yes, preferably drop the `%` compiler spec.

Also, in light of the `@` specs, you should not need separate ``version:`` tags, see  http://spack.readthedocs.io/en/latest/build_settings.html#external-packages . I understand those two to be available for orthogonal needs.

Reply:
PS: The "no-module config" you suggested for `intelpackage.rst` is from your setup, with MPI working, correct?
``` yaml
intel-mpi:
    paths:
      intel-mpi@2018.2.199: /apps/intel/ComposerXE2018/
```
I ask (as a sanity check) if the `paths:` tag is sufficiently specific for `intel-mpi`.